### PR TITLE
BlockRegion supported compressed file

### PR DESCRIPTION
**Pull request by kkvskkkk**

### CODE REVIEW

**Comment by LiXizhi** [comment]
looks good to me.  is this fully tested? 

i already wrapped zip/unzip in to a helper class, which you may use directly

https://github.com/LiXizhi/NPLRuntime/wiki/FilesAndIO#compress-and-decompress-with-zlib-and-gzip

File: Client/trunk/ParaEngineClient/BlockEngine/BlockRegion.cpp [hunk 1]
```diff
@@ -899,172 +905,206 @@ namespace ParaEngine
...
-			return;
-		}
+					stream.zalloc = (alloc_func)0;
+					stream.zfree = (free_func)0;
+					stream.opaque = (voidpf)0;
 
-		uint32_t fileTypeId = pFile->ReadDWORD();
-		if(fileTypeId != 0x626c6f63)
-			return;
+					int err = deflateInit(&stream, Z_DEFAULT_COMPRESSION);
```
Comment (LiXizhi) [inline Client/trunk/ParaEngineClient/BlockEngine/BlockRegion.cpp]: i already wrapped zip/unzip in to a helper class, which you may use directly

https://github.com/LiXizhi/NPLRuntime/wiki/FilesAndIO#compress-and-decompress-with-zlib-and-gzip

**Comment by kkvskkkk** [comment]
好， 我明天一起提交

**Comment by Xuntian** [comment]
@kkvskkkk 我在linux下编译nplruntime时遇到错误，
/NPLRuntime/Client/trunk/ParaEngineClient/IO/ZipArchive.cpp:71:10: error: converting to 'bool' from 'std::nullptr_t' requires direct-initialization [-fpermissive]
   return nullptr;

**Comment by kkvskkkk** [comment]
@LiXizhi @Xuntian 编译问题已修复， 发在这里了
https://github.com/LiXizhi/NPLRuntime/pull/86

**Pull request merged**